src/TropicalGeometry/matrix.jl

```diff
@@ -38,6 +38,6 @@ function det(A::Generic.MatSpaceElem{<:TropicalSemiringElem})
     return detA
 end
 
-function det(A::Matrix{TropicalSemiringElem{minOrMax}}) where {minOrMax<:Union{typeof(min),typeof(max)}}
+function det(A::Matrix{R}) where {R<:Union{TropicalSemiringElem,MPolyRingElem{<:TropicalSemiringElem},PolyRingElem{<:TropicalSemiringElem}}}
     return det(matrix(TropicalSemiring{minOrMax},A))
```

Comment:
This coercion won't work for polys and mpolys.
Could you add a test with a julia matrix of tropical polys exercising this?

Reply:
True and good catch.  I've tried to fix it, let's see whether it passes the tests.